packages/tldraw/src/lib/shapes/shared/useAsset.ts

```diff
@@ -0,0 +1,25 @@
+import { TLAssetId, debounce, useEditor, useValue } from '@tldraw/editor'
+import { useEffect, useRef, useState } from 'react'
+
+/** @internal */
+export function useAsset(assetId: TLAssetId | null) {
+	const editor = useEditor()
+	const [url, setUrl] = useState('')
+	const asset = assetId ? editor.getAsset(assetId) : null
+
+	// We debounce the zoom level to reduce the number of times we fetch a new image and,
+	// more importantly, to not cause zooming in and out to feel janky.
+	const [debouncedZoom, setDebouncedZoom] = useState(editor.getZoomLevel())
+	const zoomUpdater = useRef(debounce((zoom: number) => setDebouncedZoom(zoom), 500))
+	useValue('zoom level', () => zoomUpdater.current(editor.getZoomLevel()), [editor])
+
+	useEffect(() => {
+		async function resolve() {
+			const resolvedUrl = await editor.resolveAssetUrl(assetId, { zoom: debouncedZoom })
```

Comment:
what do you think about having this return a pre-multiplied (ie already transformed to the nearest LOD) zoom/scale value? 

I think there's a fair bit of nuance around selecting the right scale for zoom factor LODs - multiplicative vs linear scaling (ie powers of 2 instead of multiples of 0.25), rounding up vs rounding down, etc.

I wonder if we're possibly better placed than implementers to make those decisions & get that right whilst avoiding e.g. accidentally requesting a new image for every random fractional zoom level a user ends on. 

I think I'd be tempted here to pass the pre-rounded values to this function in the name of having devs fall into the pit of success. We can offer options for more fine-grained scaling down the line if we see a need for it. what do you think?

Reply:
definitely in agreement to create a pit of success and make it easier for implementers, that sounds fab.

i might be confused on where you're suggesting it should be pre-multiplied+returned. via `resolveAssetUrl`? or, are you saying some of the logic in `assetHandler` should be moved here? i definitely would be happier to have more of that logic live in the tldraw package! i guess i thought you wanted more of that stuff _out_ of the packages and less imposed upon the implementer :P

Reply:
i was suggesting that rather than just passing the zoom/scale/whatever to `onResolveAssetUrl`, we pass the version that's already been rounded to whatever scale we think makes sense

Reply:
ok, makes sense, gotcha. 
in that case, i wonder then if we should send two zooms : `steppedZoom` and `rawZoom`. i could see (possibly) some case where people need more fidelity then a stepped zoom.